Lab_1-2-3/lab-1-2-3-sql-window-function.md

```diff
@@ -1306,222 +1367,207 @@ Zbiór wynikowy powinien zawierać:
 - wartość sprzedaży produktu narastające od początku miesiąca
 
 ```sql
-SELECT
+with Data as (
+    SELECT
     id,
     productid,
     date,
-    value,
-    SUM(value) OVER (PARTITION BY productid, YEAR(date), MONTH(date) ORDER BY date ROWS BETWEEN UNBOUNDED PRECEDING AND CURRENT ROW) AS cumulative_value
-    from product_history
-ORDER by
-	productid,
-    YEAR(date),
-    MONTH(date),
-    date;
+    sum(unitprice*quantity) over(partition by productid,convert(date,date)) dayValue
+    FROM product_history
+)
+
+SELECT distinct
+    d.*,
+    sum(d.dayValue) over(partition by d.productid, year(d.date), month(d.date)
+    order by day(d.date) rows between unbounded preceding and current row) as accumulated
+FROM Data as d
+ORDER BY d.date
 ```
 
 **Wynik**
 
-| Postgres                      |
-| ----------------------------- |
-| ![](./img/ex14/postgres1.png) |
+![](./img/ex14/mssql1.png)
```

Comment:
Trzeba todać informację o tym, że są powtórki. mamy po 3 takie same produkty na każdy dzień i nie wynika to z błędu tylko tego, że tabela product_history została utworzona z takim błędem. Ale logika zapytania pozostaje taka sama